### DISCUSSION TITLE
Error fitting model in Keras

### DISCUSSION THREAD

**Discussion by wildaan24** [teamalgoritma/community#931] [Q&A] [answered]
Halo Algoritma team, 

saya menemukan error ketika run fitting model history untuk melihat visualnya 

![image](https://github.com/user-attachments/assets/ed2202c0-4eb7-4467-b86d-4255f871aac9)

ketika di kelas, disarankan untuk run ulang. Tapi beberapa kali saya run ulang, hasilnya tetap error. Mohon penjelasanya. Thank you

**Comment by wildaan24** [comment]
setelah saya lihat lagi, saya mendapati error pada saat run one-hot-encoding 
![image](https://github.com/user-attachments/assets/d540a80c-c49e-45a3-9f18-2089d302b1d8)

tetapi ketika saya hapus numClasses nya , syntaxnya berjalan
![image](https://github.com/user-attachments/assets/2062569d-b530-48f9-9ac0-44d0dd4aef97)

**Comment by hwulanayu** [comment]
Dear @wildaan24,

Sepertinya error terjadi karena Bapak menuliskan parameter `num_classes = 10 `sedangkan keras menerima ada 24 kelas unik (`ValueError: Shapes (None, 1) and (None, 24) are incompatible`). Pak Wildan dapat mencoba mengubah value num_classes menjadi 24. (`num_classes = 24`). Jika masih error boleh diinfokan saja yaa

**Reply by wildaan24** [reply]
halo mba wulan, 

thanks untuk feedbacknya. 
Setelah saya ganti num_classesnya menjadi 24, errornya masih tetap sama, Mbak. 
Mungkin saya coba perlihatkan syntax untuk arsitekturnya dulu ya mba :

![image](https://github.com/user-attachments/assets/41a97814-7ffc-4533-8712-9bb3109b4c16)
![Uploading image.png…]()

**Reply by wildaan24** [reply]
![image](https://github.com/user-attachments/assets/8d918e85-2596-4b3c-a5e7-24e3ebac2a67)

**Reply by hwulanayu** [reply, marked as answer]
Baik, Jika masih tidak bisa, Bapak dapat cek dimensi tiap data inputnya, apakah sudah sesuai atau belum.

```
dim(train_x) 
dim(test_x)
dim(train_y) 
dim(test_y)
```

Pastikan dimensi input data sesuai dengan input_num dan tidak bernilai `NULL`. Jika masih errror, boleh disampaikan kembali.

Additional:

Silakan ganti value parameter `x` dan `y` di fungsi `fit()` dengan variabel input terakhir yang sudah dalam bentuk array (data train) dan sudah di-one-hot-encoding (data test)

**Reply by IrfanChairurrachman** [reply]
Dear @wildaan24 

Terima kasih telah bertanya melalui Github discussions Algoritma. Untuk menjaga agar forum diskusi berjalan kondusif, karena tidak ada respon lanjutan setelah 7 hari dari respon terakhir, kami akan menandai pertanyaan sudah terjawab dan menutup diskusi ini. Semoga dapat dipahami dan terima kasih.

Best regards,
Team Algoritma
